cooker/cooker.py

```diff
@@ -771,15 +771,22 @@ def run_bitbake(self, build_config, bb_task, bb_target):
         if complete.returncode != 0:
             fatal_error('Execution of {} failed.'.format(command_line))
 
-    def shell(self, build_names: List[str]):
+    def shell(self, build_names: List[str], cmd: str = None):
```

Comment:
with `from typing import List, Optional` on top.

```suggestion
    def shell(self, build_names: List[str], cmd: Optional[List[str]] = None):
```